### ISSUE TITLE
Overfit? Validation loss barely declines while training loss decreases

### ISSUE THREAD

**Issue by CiSong10** [PatBall1/detectree2#213]
Dear developer,

When training the model and plotting the training and validation losses as described in the tutorial, I noticed that, although my training loss declines, the validation loss barely does. This very much differs from the sample [image](https://patball1.github.io/detectree2/_images/train_val_loss.png) in the tutorial, where validation loss decreases alongside training loss. 

Below is the model I trained for a large city, with a mixed urban and forest landscape. I understand the model is originally intended for forest scenarios, so I annotated a large number of training tiles (608 100×100 m² tiles). It is trained without a pre-trained model.

<img width="1920" height="1440" alt="Image" src="https://github.com/user-attachments/assets/39ecccaf-1fc4-41dc-92c1-bbe438a34d4d" />

<img width="1920" height="1440" alt="Image" src="https://github.com/user-attachments/assets/a7556a36-5089-43e1-aef3-e1ca089d2f8a" />

We can see that training loss dropped from ~3.5 to below 2, while validation loss plateaued around 2.25. Interestingly, the validation AP50 still improves slowly for all training sites after ~1000 iterations, despite fluctuations.

To check if the issue is caused by complex landscapes, I trained on just one site, a green urban area, with large and easy to delienate trees., hoping the metrics would be better. I used the pre-trained model `250312_flexi.pth` and tried all three different augmentation resize method:

<img width="1920" height="1440" alt="Image" src="https://github.com/user-attachments/assets/5777fd9d-24ef-449a-a724-cdcf8a88edad" />

Same issue (but worser), there are clear gaps between validation loss and training loss. 

Does similar thing happen to you before, and do you have any idea why this happened? Could this indicate a problem with my data, code, or training strategy? Any guidance or suggestions would be greatly appreciated!
